### PR TITLE
Update: Constant padding bottom (fixes #81)

### LINKED ISSUE #81: Timeline X Axis label padding has changed
**Describe the bug**
X Axis label has moved down and getting cut-off

**To Reproduce**

1. Go to https://engineering.cerner.com/carbon-graphs/#/timeline/simple

**Expected behavior**

Label should be clearly visible

**Screenshots**

image

**Desktop (please complete the following information):**

-   OS: [e.g. iOS] Windows 10
-   Browser [e.g. chrome, safari] Chrome

### PR DESCRIPTION
**Pull request by pranav300**
Fixes #81

**What is the purpose of this pull request? (Check any that's applicable)**

-   [ ] Feature
-   [x] Bug fix
-   [ ] Dependency updates
-   [ ] Documentation updates
-   [ ] Build related changes
-   [ ] Changes an existing functionality
-   [ ] Other, please explain:

**Does this introduce a breaking change?**

-   [ ] Yes
-   [x] No

**What changes did you make? (Give an overview)**
Updated constant padding-bottom to be 2.5
**Add any screenshots**
OS: Windows
Browser: Chrome

Before:
chrome before

After:
chrome after

Browser: Firefox

Before: 
firefox before

After:
firefox after

OS: Mac
Browser: Chrome

Before:
mac chrome before

After:
mac chrome after

### CODE REVIEW

**Comment by avinashg1994** [comment]
Changes look good, but can we also get this verified with the supported browsers mentioned [here](https://github.com/cerner/carbon-graphs#browser-support)

File: src/main/js/helpers/axis.js [hunk 1]
```diff
@@ -641,7 +641,16 @@ const getAxisLabelHeight = (label) => {
...
     const grouper = svg.append("g");
     buildAxisLabel(grouper, label);
-    const height = grouper.node().getBoundingClientRect().height;
+    let height = 0;
+    if (window.navigator.platform === "Win32") {
+        if (window.navigator.userAgent.indexOf("Firefox") > -1) {
+            height = grouper.node().getBoundingClientRect().height - 3.5;
+        } else {
+            height = grouper.node().getBoundingClientRect().height - 2;
+        }
```
Comment (abhijit945) [inline src/main/js/helpers/axis.js]: Can we not introduce any browser specific code? I would like to stay away from browser/OS level code. 
If the fix can be resolved throughout the OSes and Browsers by increasing the padding then I think that is justification enough for us to do that.

Comment (pranav300) [reply src/main/js/helpers/axis.js]: Fixed here https://github.com/cerner/carbon-graphs/pull/107/commits/19c7116da55ad4aaf689cbe1cb28abd648c1139e

File: src/main/js/helpers/axis.js [hunk 2]
```diff
@@ -645,7 +645,8 @@ const getAxisLabelHeight = (label) => {
     const svg = dummy.append("svg");
     const grouper = svg.append("g");
     buildAxisLabel(grouper, label);
-    const height = grouper.node().getBoundingClientRect().height;
+    let height = 0;
```
Comment (abhijit945) [inline src/main/js/helpers/axis.js]: Is this doing anything?

File: src/main/js/helpers/constants.js [hunk 3]
```diff
@@ -149,7 +149,7 @@ export const AXES_ORIENTATION = {
 export default {
     PADDING: {
         top: 10,
-        bottom: 5,
+        bottom: 2.5,
```
Comment (abhijit945) [inline src/main/js/helpers/constants.js]: Why would reducing the bottom padding fix this?

File: src/main/js/helpers/axis.js (hunk 2 above)
Comment (pranav300) [reply src/main/js/helpers/axis.js]: No, that can be changed back to const.

File: src/main/js/helpers/constants.js (hunk 3 above)
Comment (pranav300) [reply src/main/js/helpers/constants.js]: So, bottom padding pushes the graph from the bottom towards the top. Hence by reducing it, we stop pushing the graph towards the top. And the previous solution suggested by @ShettyAkarsh i.e `(config.padding.top) * 2 + config.padding.bottom;` does the same as reducing the bottom padding by half. And this won't even affect the consistency between the graphs.

**Comment by abhijit945** [comment]
Please add and update unit tests. 
The build is failing as well.

File: src/test/unit/controls/PairedResult/PairedResultPanning-spec.js [hunk 4]
```diff
@@ -98,7 +98,7 @@ describe("PairedResult", () => {
                     datelineGroup.getAttribute("transform")
                 ).translate;
                 expect(toNumber(translate[0], 10)).toBeGreaterThanOrEqual(80);
-                expect(toNumber(translate[1], 10)).toBeCloseTo(5);
+                expect(toNumber(translate[1], 10)).toBeCloseTo(2);
```
Comment (abhijit945) [inline src/test/unit/controls/PairedResult/PairedResultPanning-spec.js]: Is this same for all the graph types? Maybe we can create a constant in commonHelpers and then reference that?

Comment (pranav300) [reply src/test/unit/controls/PairedResult/PairedResultPanning-spec.js]: Added here https://github.com/cerner/carbon-graphs/pull/107/commits/f960f072648a74aeb7a781ed0bc5cf91020a16e6

Comment (abhijit945) [reply src/test/unit/controls/PairedResult/PairedResultPanning-spec.js]: I meant in the `commonHelpers` [here](https://github.com/cerner/carbon-graphs/blob/master/src/test/unit/helpers/commonHelpers.js)

Comment (pranav300) [reply src/test/unit/controls/PairedResult/PairedResultPanning-spec.js]: Done here https://github.com/cerner/carbon-graphs/pull/107/commits/f1b827cc6c61574d07c286ad6744c5da18bb9e90.

**Comment by abhijit945** [comment]
Is this good to be merged? 
Does this fix https://github.com/cerner/carbon-graphs/issues/117?

**Comment by pranav300** [comment]
> Is this good to be merged?

  Yes

> Does this fix #117?

No, While investigating the issue I found that we are using `top` and `bottom` padding to get the x-axis width. But I don't understand why? Check here [creationHelper](https://github.com/cerner/carbon-graphs/blob/master/src/main/js/controls/Timeline/helpers/creationHelpers.js#L93-L107)
According to me the `top` and `bottom` padding should not be considered for width.

File: src/test/unit/helpers/commonHelpers.js [hunk 5]
```diff
@@ -3,6 +3,10 @@ import d3 from "d3";
  * @type {number} Delay duration
  */
 export const TRANSITION_DELAY = 400;
+/*
+ * @type {number} Padding bottom
+ */
+export const PADDING_BOTTOM = 2;
```
Comment (ShettyAkarsh) [inline src/test/unit/helpers/commonHelpers.js]: should this be set to  2.5 ?

**Pull request merged**